tests/unit/test_dns_client.cpp

```diff
@@ -1723,13 +1811,204 @@ void TestDnssdServerProxyCallback(void)
     Log("End of TestDnssdServerProxyCallback");
 }
 
+void TestDnssdSoaNsResponse(void)
+{
+    Srp::Server      *srpServer;
+    Srp::Client      *srpClient;
+    Dns::Client      *dnsClient;
+    Dns::Name::Buffer serverName;
+    StringWriter      writer(serverName, sizeof(serverName));
+
+    Log("--------------------------------------------------------------------------------------------");
+    Log("TestDnssdSoaNsResponse");
+
+    InitTest();
+
+    srpServer = &sInstance->Get<Srp::Server>();
+    srpClient = &sInstance->Get<Srp::Client>();
+    dnsClient = &sInstance->Get<Dns::Client>();
+
+    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    // Start SRP server.
+
+    SuccessOrQuit(srpServer->SetAddressMode(Srp::Server::kAddressModeUnicast));
+    VerifyOrQuit(srpServer->GetState() == Srp::Server::kStateDisabled);
+
+    srpServer->SetEnabled(true);
+    VerifyOrQuit(srpServer->GetState() != Srp::Server::kStateDisabled);
+
+    AdvanceTime(10000);
+    VerifyOrQuit(srpServer->GetState() == Srp::Server::kStateRunning);
+
+    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    // Start SRP client.
+
+    srpClient->EnableAutoStartMode(nullptr, nullptr);
+    VerifyOrQuit(srpClient->IsAutoStartModeEnabled());
+
+    AdvanceTime(2000);
+    VerifyOrQuit(srpClient->IsRunning());
+
+    writer.Append("otDNS%s.default.service.arpa.", sInstance->Get<Mac::Mac>().GetExtAddress().ToString().AsCString());
+
+    Log("- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - ");
+    Log("SOA Query");
+
+    for (uint8_t iter = 0; iter < 2; iter++)
+    {
+        sQueryRecordInfo.Reset();
+
+        // First iteration: Query for `default.service.arpa.` directly, and
+        // validate that we see the SOA response in the Answer section.
+        // Second iteration: Query for `myhost.default.service.arpa.`, which
```

Comment:
See prev comment about no subdomains - suggested to return NXDOMAIN, 0 answers and SOA in authority section.

Reply:
Correction:  RCODE=NoError, 0 answers, 1 SOA record in Authority section.
(this is the correct response when a DP is active - we'll just ignore the other case where DP is not active to keep it simple - devices without DP aren't Thread certified anyway).